app/service/planning_svc.py

```diff
@@ -50,8 +54,32 @@ def __init__(self):
                 links.extend(await self._check_and_generate_cleanup_links(agent, operation, link_status))
         return reversed(await self.trim_links(operation, links, agent))
 
+    async def check_stopping_conditions(self, operation, stopping_conditions):
+        """
+        Checks whether an operation has collected the proper facts to trigger this planner's stopping
+        conditions
+        :param operation:
+        :param stopping_conditions:
+        :return: True if all stopping conditions have been met, False if all stopping conditions have not
+        been met
+        """
+        if len(stopping_conditions) < 1:
+            return False
+        for sc in stopping_conditions:
+            if not await self._stopping_condition_met(operation.all_facts(), sc):
+                return False
+        return True
+
     """ PRIVATE """
 
+    @staticmethod
+    async def _stopping_condition_met(facts, stopping_condition):
+        for f in facts:
+            items = list(stopping_condition.items())[0]
```

Comment:
should stopping_condition(s) be converted into a Fact() object on the c_planner? I think that would be the right move here. thoughts?

Reply:
I believe that feature is coming in the next step of this PR.  Right now, we are _only_ supporting items listed in the planner params.  There is another ticket to allow user to define new stop conditions from the UI, which will include a transition to Facts

Reply:
Brian can provide more details though

Reply:
cool - that makes sense. I'd like to see stopping conditions go in as a fact in this PR (unless there's a good reason not to). since it's a new property - might as well put it into the right type right away.

Reply:
@brianedmonds90 if you're not aware - a new feature dropped today, showing planners from the GUI (advanced tab). if you end up adding edit functionality to the planners, this would be the spot.

Reply:
Roger that